mobileApp/jobs/buildWhiteLabelAndroidApps.groovy

```diff
@@ -3,24 +3,44 @@ package mobile
 import org.yaml.snakeyaml.Yaml
 import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
 
-/*
-Example secret YAML file used by this script
-jobConfig:
-    jobName: build-demo-android-app
-    public: true
-    release: false => whether or not the job will build a release apk (as opposed to a debuggable apk)
-    buildScript: build.sh
-    gradleProperties: GRADLE_PROPERTIES
-    keystore: APP_KEYSTORE
-    keystoreFile: APP_KEYSTORE_FILE
-    gitRepo: https://github.com/org/project.git
-    gitCredential: git_user
-    appBaseDir: android-app-project
-    appBaseName: edx-demo-app
-    generatedApkName: edx-demo-app.apk
-    sshAgent: ssh_user
-    hockeyAppApiToken: abc123
-*/
+/**
+ buildWhiteLabelAndroidApps.groovy
+
+ This dsl-script creates Jenkins jobs to perform the following:
+   - Build an apk based on a user-supplied git-hash from a given edx android repository
+   - Archive the apk file on Jenkins for history & quick testing
+   - Publish the apk to HockeyApp (a service for hosting & disrtibuting application files)
+   - TODO: Run tests on the generated apk file (linting, unit tests, screenshot tests)
+ Depending on the "release" flag mentioned below, the job will be considered either 'release' or 'debug'
+ Release builds will also use special gradle.properties and keystore files when compiled, to sign
+ the build.
+
+ ==========================================================================================
+ In order to run, the job requires the following be loaded into Jenkins as secrets files:
+   - A secret yml file: This contains a number of jobConfigs, one for each desired Jenkins job
+   - A gradle.properties file: A key-value file used in the gradle build process **
```

Comment:
I think we do need two gradle properties, but there's just no keystore for debug builds (there's a ** footnote below).